mindsdb_native/libs/backends/lightwood.py

```diff
@@ -340,7 +350,12 @@ def train(self):
 
                 kwargs['callback_on_iter'] = self.callback_on_iter
                 kwargs['eval_every_x_epochs'] = eval_every_x_epochs / len(mixer_classes)
-                kwargs['stop_training_after_seconds'] = self.transaction.lmd['stop_training_in_x_seconds']
+                
+                if self.transaction.lmd['stop_training_in_x_seconds'] is not None:
+                    if self.transaction.lmd['stop_training_in_x_seconds'] > reasonable_training_time:
+                        kwargs['stop_training_after_seconds'] = self.transaction.lmd['stop_training_in_x_seconds'] / 2
+                    elif reasonable_training_time / 10 < self.transaction.lmd['stop_training_in_x_seconds'] < reasonable_training_time:
+                        kwargs['stop_training_after_seconds'] = self.transaction.lmd['stop_training_in_x_seconds']
```

Comment:
This logic should happen during the previous if/else chain when the mixers are selected